src/Compilers/Core/Portable/StrongName/PortableStrongNameProvider.cs

```diff
@@ -0,0 +1,90 @@
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class PortableStrongNameProvider : StrongNameProvider
+    {
+        private readonly ImmutableArray<string> _keyFileSearchPaths;
+        internal StrongNameFileSystem FileSystem { get; }
+
+        public PortableStrongNameProvider(ImmutableArray<string> keySearchPaths, StrongNameFileSystem strongNameFileSystem)
+        {
+            FileSystem = strongNameFileSystem ?? StrongNameFileSystem.Instance;
+            _keyFileSearchPaths = keySearchPaths.NullToEmpty();
+        }
+
+        public override int GetHashCode()
+        {
+            return 0;
+        }
+
+        internal override SigningCapability Capability => SigningCapability.SignsPeBuilder;
+
+        internal override StrongNameKeys CreateKeys(string keyFilePath, string keyContainerName, CommonMessageProvider messageProvider)
+        {
+            var keyPair = default(ImmutableArray<byte>);
+            var publicKey = default(ImmutableArray<byte>);
+            string container = null;
+
+            if (!string.IsNullOrEmpty(keyFilePath))
+            {
+                try
+                {
+                    string resolvedKeyFile = FileSystem.ResolveStrongNameKeyFile(keyFilePath, _keyFileSearchPaths);
+                    if (resolvedKeyFile == null)
+                    {
+                        var message = $"{keyFilePath}: {CodeAnalysisResources.FileNotFound}";
+                        return new StrongNameKeys(StrongNameKeys.GetKeyFileError(messageProvider, keyFilePath, message));
+                    }
+
+                    Debug.Assert(PathUtilities.IsAbsolute(resolvedKeyFile));
+                    var fileContent = ImmutableArray.Create(FileSystem.ReadAllBytes(resolvedKeyFile));
+                    return StrongNameKeys.CreateHelper(fileContent, keyFilePath);
+                }
+                catch (Exception ex)
+                {
+                    return new StrongNameKeys(StrongNameKeys.GetKeyFileError(messageProvider, keyFilePath, ex.Message));
+                }
+                // it turns out that we don't need IClrStrongName to retrieve a key file,
+                // so there's no need for a catch of ClrStrongNameMissingException in this case
+            }
+
+            return new StrongNameKeys(keyPair, publicKey, null, container, keyFilePath);
+        }
+
+        internal override void SignPeBuilder(ExtendedPEBuilder peBuilder, BlobBuilder peBlob, RSAParameters privkey)
+        {
+            peBuilder.Sign(peBlob, content => SigningUtilities.CalculateRsaSignature(content, privkey));
+        }
+
+        internal override Stream CreateInputStream()
+        {
+            throw new NotSupportedException();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is null || GetType() != obj.GetType())
+            {
+                return false;
+            }
+
+            var other = (PortableStrongNameProvider)obj;
```

Comment:
This is a `sealed` type so the `GetType()` comparison is unnecessary. Consider simplifying to:
```
if (!(obj is PortableStrongNameProvider other))
{
    return false;
}
```

Reply:
Done.